### ISSUE TITLE
Opening an invalid direct link triggers a error exception page

### ISSUE THREAD

**Issue by LukasReschke** [nextcloud/server#952]
To reproduce access an not existing invalid direct link such as https://cloud.nextcloud.com/f/1357

```
Internal Server Error
The server encountered an internal error and was unable to complete your request.
Please contact the server administrator if this error reappears multiple times, please include the technical details below in your report.
More details can be found in the server log.

Technical details
Remote Address: 81.62.83.43
Request ID: d7uNkztsOKoW+weoXuht
Type: OCP\Files\NotFoundException
Code: 0
Message:
File: /var/www/html/apps/files/lib/Controller/ViewController.php
Line: 309

Trace
#0 [internal function]: OCA\Files\Controller\ViewController->showFile('1357')
#1 /var/www/html/lib/private/AppFramework/Http/Dispatcher.php(160): call_user_func_array(Array, Array)
#2 /var/www/html/lib/private/AppFramework/Http/Dispatcher.php(90): OC\AppFramework\Http\Dispatcher->executeController(Object(OCA\Files\Controller\ViewController), 'showFile')
#3 /var/www/html/lib/private/AppFramework/App.php(111): OC\AppFramework\Http\Dispatcher->dispatch(Object(OCA\Files\Controller\ViewController), 'showFile')
#4 /var/www/html/lib/public/AppFramework/App.php(136): OC\AppFramework\App::main('ViewController', 'showFile', Object(OC\AppFramework\DependencyInjection\DIContainer))
#5 /var/www/html/core/routes.php(122): OCP\AppFramework\App->dispatch('ViewController', 'showFile')
#6 [internal function]: OC\Route\Router->{closure}(Array)
#7 /var/www/html/lib/private/Route/Router.php(293): call_user_func(Object(Closure), Array)
#8 /var/www/html/lib/base.php(967): OC\Route\Router->match('/f/1357')
#9 /var/www/html/index.php(48): OC::handleRequest()
#10 {main}
```

This should be caught instead and a 404 be displayed instead of spamming the server log.

**Comment by rullzer** [comment]
Yes agreed. I'll take care of this.

**Comment by rullzer** [comment]
Do we want to show the actual 404 page or just redirect to files?

**Comment by rullzer** [comment]
Fix is in #956
